app/src/main/java/in/ac/bits_pilani/goa/ard/utils/Browser.java

```diff
@@ -0,0 +1,408 @@
+     * @param exitResId Resource ID of the "exit" animation for the browser.
+     * @return current instance.
+     */
+    public Browser setExitAnimations(@AnimRes final int enterResId, @AnimRes final int exitResId) {
+        this.exitAnimEnterResId = enterResId;
+        this.exitAnimExitResId = exitResId;
+        return this;
+    }
+
+    /**
+     * Don't hide url bar when user scrolls.
+     * @return current instance.
+     */
+    public Browser disableUrlBarHiding() {
+        urlBarHiding = false;
+        return this;
+    }
+
+    /**
+     * Hide url bar when user scrolls (default).
+     * @return current instance.
+     */
+    public Browser enableUrlBarHiding() {
+        urlBarHiding = true;
+        return this;
+    }
+
+    /**
+     * Don't show title of web-page.
+     * @return current instance.
+     */
+    public Browser disableTitle() {
+        titleEnabled = false;
+        return this;
+    }
+
+    /**
+     * Show title of web-page (default).
+     * @return current instance.
+     */
+    public Browser enableTitle() {
+        titleEnabled = true;
+        return this;
+    }
+
+    /**
+     * Adds a menu item.
+     * @param label Menu label.
+     * @param pendingIntent Pending intent delivered when the menu item is clicked.
+     * @return current instance.
+     */
+    public Browser addMenuItem(@NonNull final String label, @NonNull final PendingIntent pendingIntent) {
+        if (menuItems == null) {
+            menuItems = new ArrayList<>();
+        }
+        final Bundle bundle = new Bundle();
+        bundle.putString(KEY_MENU_ITEM_TITLE, label);
+        bundle.putParcelable(KEY_PENDING_INTENT, pendingIntent);
+        menuItems.add(bundle);
+        return this;
+    }
+
+    /**
+     * Enable share item in menu.
+     * @return current instance.
+     */
+    public Browser enableShareInMenu() {
+        shareInMenu = true;
+        return this;
+    }
+
+    /**
+     * Disable share item in menu.
+     * @return current instance.
+     */
+    public Browser disableShareInMenu() {
+        shareInMenu = false;
+        return this;
+    }
+
+    /**
+     * Disable action button in toolbar.
+     * @return current instance.
+     */
+    public Browser disableActionButton() {
+        actionButtonEnabled = false;
+        return this;
+    }
+
+    /**
+     * Sets the action button that is displayed in the Toolbar.
+     * @param icon The icon.
+     * @param width width of icon in dp. Should be in range [24,48], scaled corresponding to height of 24dp.
+     * @param description The description for the button. To be used for accessibility.
+     * @param pendingIntent pending intent delivered when the button is clicked.
+     * @param shouldTint Whether the action button should be tinted.
+     * @return current instance.
+     */
+    public Browser setActionButton(@NonNull final Bitmap icon,
+                                   final int width,
```

Comment:
Is width really important? it doesn't seem this will be used. @Rushi98 you may remove this param and use some default value
@yashsharan @Rushi98 any thoughts?